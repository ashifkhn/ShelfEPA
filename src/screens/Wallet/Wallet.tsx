import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';

const Wallet = () => {
  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.headerText}>Your Wallet</Text>
        <Text style={[styles.headerText, styles.primaryText]}>Total Balance: $500</Text>
      </View>
      <View style={styles.content}>
        <Text style={styles.contentText}>Manage your funds and transactions here.</Text>
        <TouchableOpacity style={styles.button}>
          <Text style={styles.buttonText}>Add Money</Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.transactionButton}>
          <Text style={styles.buttonText}>View Transactions</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    padding: 20,
  },
  header: {
    marginTop: 50,
    marginBottom: 30,
  },
  headerText: {
    fontSize: 24,
    color: '#0373F3',
    textAlign: 'center',
  },
  primaryText: {
    fontWeight: 'bold',
  },
  content: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  contentText: {
    fontSize: 18,
    color: '#0373F3',
    textAlign: 'center',
    marginBottom: 20,
  },
  button: {
    backgroundColor: '#0373F3',
    padding: 15,
    borderRadius: 5,
    marginBottom: 10,
  },
  transactionButton: {
    backgroundColor: '#0373F3',
    padding: 15,
    borderRadius: 5,
  },
  buttonText: {
    color: 'white',
    fontSize: 16,
    textAlign: 'center',
  },
});

export default Wallet;
