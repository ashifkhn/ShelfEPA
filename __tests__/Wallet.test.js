import "react-native"
import React from "react"
import renderer from "react-test-renderer"
import Wallet from "../src/screens/Wallet/Wallet";

test('Home renders correctly', () => {
  const tree = renderer.create(<Wallet />).toJSON();
  expect(tree).toMatchSnapshot();
});