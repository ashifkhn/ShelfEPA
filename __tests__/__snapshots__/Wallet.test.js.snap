// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "marginBottom": 30,
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        {
          "color": "#0373F3",
          "fontSize": 24,
          "textAlign": "center",
        }
      }
    >
      Your Wallet
    </Text>
    <Text
      style={
        [
          {
            "color": "#0373F3",
            "fontSize": 24,
            "textAlign": "center",
          },
          {
            "fontWeight": "bold",
          },
        ]
      }
    >
      Total Balance: $500
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": "#0373F3",
          "fontSize": 18,
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      Manage your funds and transactions here.
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#0373F3",
          "borderRadius": 5,
          "marginBottom": 10,
          "opacity": 1,
          "padding": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Add Money
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#0373F3",
          "borderRadius": 5,
          "opacity": 1,
          "padding": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        View Transactions
      </Text>
    </View>
  </View>
</View>
`;
